pandas/core/indexes/base.py

```diff
@@ -7161,10 +7181,12 @@ def unpack_nested_dtype(other: _IndexT) -> _IndexT:
 
 
 def _maybe_try_sort(result, sort):
-    if sort is None:
+    if sort is not False:
         try:
             result = algos.safe_sort(result)
         except TypeError as err:
+            if sort is True:
```

Comment:
Do we have tests for these "stricter" raising behavior?

Reply:
Yes, they’re xfailed in main